### PR TITLE
Allow adding solutions in short answer quiz re-evaluation

### PR DESCRIPTION
**Pull request by FuchsDominik**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

It was not possible until now to change solutions when re-evaluating short answer quiz questions. On request, this should be possible now.

### Description

- Removed the check which looks and newly added solutions
- Leaves the check which checks for the inValid state of the solution
- Tested the new behaviour with an Server Integration test

### Steps for Testing

For a normal Quiz Exercise
1. Log in to Artemis
2. Create an quiz with a short answer question
3. Participate and answer every spot correctly except for one.
4. Check that after the quiz is over, the wrong answer is detected correctly.
5. Go to Course Management, Exercises and Select Re-Evaluate on the quiz
6. Accept your wrong answer as a new solution in the spot
7. Save the re-evaluation
8. Check that as a participant, the quiz results are adapted accordingly and the wrong solution a moment ago is now counted as correct

For an exam
1. Log in to Artemis
2. Create an exam with a short answer question quiz
3. Participate and answer every spot correctly except for one.
4. After the exam is over, go to student exams and click evaluate quizzes
5. Check that the wrong answer is detected correctly. (e.g. in student exams -> view -> check points)
6. Go to the exam overview, exercise groups and click on re-evaluate quiz
7. Accept your wrong answer as a new solution in the spot
8. Save the re-evaluation
9. Check again (in student exams -> view -> check points) that the quiz results are adapted accordingly and the wrong solution a moment ago is now counted as correct

### Test Coverage

Tested changes accordingly

### CODE REVIEW

**Review by airelawaleria — APPROVED** [review]
tested on ts1, all works :)

**Review by EneaGore — APPROVED** [review]
Tested successfuly on ts1.

**Review by valeriebucher — DISMISSED** [review]
Worked for me!

**Review by JohannesWeiss — DISMISSED** [review]
Works as intended on ts1

**Review by simonzilker — DISMISSED** [review]
Works perfectly :)

**Review by marvinchimeka — DISMISSED** [review]
No issues found

File: src/main/java/de/tum/in/www1/artemis/domain/quiz/ShortAnswerQuestion.java [hunk 1]
```diff
@@ -263,6 +232,24 @@ private void undoUnallowedSpotChanges(ShortAnswerQuestion originalQuestion) {
         this.getSpots().removeAll(notAllowedAddedSpots);
     }
 
+    /**
+     * check all solutions for unset inValid states or state changes
```
Comment (derLalla) [inline src/main/java/de/tum/in/www1/artemis/domain/quiz/ShortAnswerQuestion.java:236]:
```suggestion
     * check all solutions for unset invalid states or state changes
```

File: src/main/java/de/tum/in/www1/artemis/domain/quiz/ShortAnswerQuestion.java [hunk 2]
```diff
@@ -263,6 +232,24 @@ private void undoUnallowedSpotChanges(ShortAnswerQuestion originalQuestion) {
...
+     */
+    private void checkInvalidSolutions(ShortAnswerQuestion originalQuestion) {
+        // check every solution of the question
+        for (ShortAnswerSolution solution : this.getSolutions()) {
+            // correct invalid = null to invalid = false
+            if (solution.isInvalid() == null) {
+                solution.setInvalid(false);
+            }
+            ShortAnswerSolution originalSolution = originalQuestion.findSolutionById(solution.getId());
+            // reset invalid solution if it already set to true (it's not possible to set a solution valid again)
```
Comment (derLalla) [inline src/main/java/de/tum/in/www1/artemis/domain/quiz/ShortAnswerQuestion.java:248]:
```suggestion
            // reset invalid solution if it was already set to true (it's not possible to set a solution valid again)
```

**Review by derLalla — APPROVED** [review]
I have just noted some typos, but is it up to you if you want to change them. So I am approving :)

**Review by nicolasruscher — APPROVED** [review]
Looks good!

**Review by fde312 — APPROVED** [review]
Works fine and code looks also good! Great job! 🚀

**Pull request merged**